### PR TITLE
[llvmonly] Emit unbox trampolines as 2 tables instead of a lookup function, this takes less space that the switch statement in the lookup function.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -5877,6 +5878,22 @@ mono_aot_get_unbox_arbitrary_trampoline (gpointer addr)
 	return mono_create_ftnptr (mono_domain_get (), code);
 }
 
+static int
+i32_idx_comparer (const void *key, const void *member)
+{
+	int idx1 = GPOINTER_TO_INT (key);
+	int idx2 = *(int*)member;
```
Comment (lewurm) [inline mono/mini/aot-runtime.c]: let's use `gint32*`?

File: mono/mini/aot-runtime.c [hunk 2]
```diff
@@ -5914,6 +5931,27 @@ mono_aot_get_unbox_trampoline (MonoMethod *method, gpointer addr)
 		return mono_aot_get_unbox_arbitrary_trampoline (addr);
 	}
 
+	if (amodule->info.llvm_unbox_tramp_indexes) {
+		int unbox_tramp_idx;
+
+		/* Search the llvm_unbox_tramp_indexes table using a binary search */
+		if (amodule->info.llvm_unbox_tramp_elemsize == 4) {
```
Comment (lewurm) [inline mono/mini/aot-runtime.c]: `s/4/sizeof (guint32)/`

File: mono/mini/aot-runtime.c (hunk 1 above)
Comment (vargaz) [reply mono/mini/aot-runtime.c]: Fixed.

File: mono/mini/aot-runtime.c (hunk 2 above)
Comment (vargaz) [reply mono/mini/aot-runtime.c]: Fixed.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**